File: Codes/3185-count-pairs-that-form-a-complete-day-ii.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-10-23 18:21:44
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-10-23 22:52:02
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

/*
0 0
1 23
2 22
...
11 13
12 12
*/
typedef long long ll;
class Solution {
public:
    ll countCompleteDayPairs(vector<int>& hours) {
        ll bin[24] = {0};
        for (int t : hours) {
            bin[t % 24]++;
        }
        ll ans = bin[0] * (bin[0] - 1) / 2 + bin[12] * (bin[12] - 1) / 2;
        for (int i = 1; i < 12; i++) {
            ans += bin[i] * bin[24 - i];
        }
        return ans;
    }
};

File: Codes/3185-count-pairs-that-form-a-complete-day-ii.go
/*
 * @Author: LetMeFly
 * @Date: 2024-10-23 22:57:45
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-10-23 23:00:15
 */
package main

func countCompleteDayPairs(hours []int) int64 {
    bin := make([]int64, 24)
    var ans int64
    for _, t := range hours {
        ans += bin[(24 - t % 24) % 24]
        bin[t % 24]++
    }
    return ans
}